app/models/chargeback_vm.rb

```diff
@@ -185,13 +185,18 @@ def self.vms(region)
           end
 
           tenant_name = tenant.name
-          tenant = Tenant.in_region(region).find_by(:name => tenant.name)
+          # NOTE: tenant.name has settings applied.
+          # i.e.: Tenant.where(:id => tenant.id, :name => tenant.name).exists? could be false
+          if tenant.region_id != region
+            tenant = Tenant.in_region(region).find_by(:name => tenant.name)
+          end
           if tenant.nil?
             error_message = "Unable to find tenant '#{tenant_name}' (based on tenant id '#{@options[:tenant_id]}' from default region) in region #{region}"
             _log.info("#{error_message}. Calculating chargeback costs skipped for #{@options[:tenant_id]} in region #{region}.")
-            return Vm.none
+            Vm.none
+          else
+            Vm.where(:tenant_id => tenant.subtree.select(:id))
```

Comment:
Is there a spec exercising a deep tenant tree?

Reply:
the spec points to the root tenant (that has the name changed - so it wouldn't be able to find it - I did red/green) and the vm is in the 3rd tenant down

Reply:
https://github.com/ManageIQ/manageiq/pull/22491/files#diff-357a88f9ef491cd08c43050cb0208751aa1b44f7749c3b5da0023af4911a7fd6R453

We use `Vm.create(:tenant => user_tenant.child)`
and filter on `tenant`.

---

But this commit is not for sub tenants.
There was a `@cache_vms = begin ; if x then Vm.all else return Vm.none ; end ; end`
and rubocop was complaining that the `return` would not populate `@cache_vms`. So I put in this commit.